src/backend/BackendInterface.h

```diff
@@ -173,39 +217,49 @@ class BackendInterface
     virtual void
     writeLedger(
         ripple::LedgerInfo const& ledgerInfo,
-        std::string&& ledgerHeader) = 0;
+        std::string&& ledgerHeader,
+        boost::asio::yield_context& yield) = 0;
 
-    void
-    writeLedgerObject(std::string&& key, uint32_t seq, std::string&& blob);
+    virtual void
+    writeLedgerObject(
+        std::string&& key,
+        uint32_t seq,
+        std::string&& blob,
+        boost::asio::yield_context& yield);
 
     virtual void
     writeTransaction(
         std::string&& hash,
         uint32_t seq,
         uint32_t date,
         std::string&& transaction,
-        std::string&& metadata) = 0;
+        std::string&& metadata,
+        boost::asio::yield_context& yield) = 0;
 
     virtual void
-    writeAccountTransactions(std::vector<AccountTransactionsData>&& data) = 0;
+    writeAccountTransactions(
+        std::vector<AccountTransactionsData>&& data,
+        boost::asio::yield_context& yield) = 0;
 
     virtual void
     writeSuccessor(
         std::string&& key,
         uint32_t seq,
-        std::string&& successor) = 0;
+        std::string&& successor,
+        boost::asio::yield_context& yield) = 0;
     // Tell the database we are about to begin writing data for a particular
     // ledger.
     virtual void
-    startWrites() = 0;
+    startWrites(boost::asio::yield_context& yield) const = 0;
 
     // Tell the database we have finished writing all data for a particular
     // ledger
     bool
-    finishWrites(uint32_t ledgerSequence);
+    finishWrites(uint32_t ledgerSequence, boost::asio::yield_context& yield);
 
     virtual bool
-    doOnlineDelete(uint32_t numLedgersToKeep) const = 0;
+    doOnlineDelete(uint32_t numLedgersToKeep, boost::asio::yield_context& yield)
+        const = 0;
```

Comment:
Let's get rid of the yield context for writes. Only Postgres needs it, and thus Postgres can do that internally.